software/firmware/europi.py

```diff
@@ -128,17 +128,18 @@ class AnalogueReader:
     not need to be used by user scripts.
     """
 
-    def __init__(self, pin, samples=DEFAULT_SAMPLES):
+    def __init__(self, pin, samples=DEFAULT_SAMPLES, deadzone=0.0):

```

Comment:
[discussion] I wonder if it might be more user friendly to move `deadzone` to be a parameter to the `percent()` method. It feels very similar to `samples` to me. I could see scripts wanting to tweak the size of the zone. This would make the API more consistent, with the parameters that affect the reading of knob position being configurable at read time. And it would make the suggestions for changes to the tests that I made in a separate comment a little easier as well.